tests/test_connection_monitor.c

```diff
@@ -284,12 +284,29 @@ static struct http_monitor_test_stats_event s_test_rw_above_events[] = {
         .expected_throughput = 1000,
     },
 };
+
+struct observer_cb_data {
+    bool invoked;
+    size_t nonce;
+};
+static void s_observer_cb(size_t connection_nonce, const struct aws_array_list *stats, void *user_data) {
+    (void)stats;
+    struct observer_cb_data *cb_data = user_data;
+    cb_data->invoked = true;
+    cb_data->nonce = connection_nonce;
+}
 static int s_test_http_connection_monitor_rw_above(struct aws_allocator *allocator, void *ctx) {
     (void)ctx;
 
+    struct observer_cb_data cb_data;
+    AWS_ZERO_STRUCT(cb_data);
+    s_test_options.statistics_observer_fn = s_observer_cb;
+    s_test_options.statistics_observer_user_data = &cb_data;
     int result = s_do_http_monitoring_test(
         allocator, &s_test_options, s_test_rw_above_events, AWS_ARRAY_SIZE(s_test_rw_above_events));
     ASSERT_TRUE(result == AWS_OP_SUCCESS);
+    ASSERT_TRUE(cb_data.invoked);
+    ASSERT_TRUE(cb_data.nonce > 0);
```

Comment:
Is there a test somewhere that tests whether or not the contents of the reported metrics are correct?

Reply:
there is now.